catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/impl/operations/ResourceOperationsTest.java

```diff
@@ -0,0 +1,500 @@
+
+    Map<String, Serializable> requestPropertiesMock;
+
+    MetacardImpl metacard;
+
+    QueryRequest queryRequestMock;
+
+    ReliableResourceDownloadManager reliableResourceDownloadManagerMock;
+
+    ResourceResponse resourceResponseMock;
+
+    @Before
+    public void setUp() throws Exception {
+        setUpFrameworkProperties();
+        setUpResourceOperationsAndMocks();
+    }
+
+    private void setUpResourceOperationsAndMocks() throws URISyntaxException {
+        queryOperationsMock = mock(QueryOperations.class);
+        queryResponseMock = mock(QueryResponse.class);
+        queryRequestMock = mock(QueryRequest.class);
+        resourceRequestMock = mock(ResourceRequest.class);
+        reliableResourceDownloadManagerMock = mock(ReliableResourceDownloadManager.class);
+        resourceResponseMock = mock(ResourceResponse.class);
+
+        testUri = new URI("bobUri", "test", "fragment");
+
+        requestPropertiesMock = mock(Map.class);
+
+        metacard = new MetacardImpl();
+        metacard.setId("Bobbert");
+        metacard.setTitle("Bobbert's Title");
+
+        resourceOperations = new ResourceOperations(frameworkProperties,
+                queryOperationsMock,
+                opsSecuritySupport);
+    }
+
+    private void setUpFrameworkProperties() {
+        frameworkProperties = new FrameworkProperties();
+        frameworkProperties.setAccessPlugins(new ArrayList<>());
+        frameworkProperties.setPolicyPlugins(new ArrayList<>());
+        frameworkProperties.setSourcePoller(mockPoller);
+        frameworkProperties.setCatalogProviders(Collections.singletonList((CatalogProvider) provider));
+        frameworkProperties.setPostResource(mockPostResourcePlugins);
+        frameworkProperties.setFederationStrategy(mockFederationStrategy);
+        frameworkProperties.setFilterBuilder(new GeotoolsFilterBuilder());
+        frameworkProperties.setPreIngest(new ArrayList<>());
+        frameworkProperties.setPostIngest(postIngestPlugins);
+        frameworkProperties.setPreQuery(new ArrayList<>());
+        frameworkProperties.setPostQuery(new ArrayList<>());
+        frameworkProperties.setPreResource(new ArrayList<>());
+        frameworkProperties.setPostResource(new ArrayList<>());
+        frameworkProperties.setQueryResponsePostProcessor(mock(QueryResponsePostProcessor.class));
+        frameworkProperties.setStorageProviders(Collections.singletonList(storageProvider));
+        frameworkProperties.setMimeTypeMapper(new MimeTypeMapperImpl(Collections.singletonList(
+                mimeTypeResolver)));
+        frameworkProperties.setMimeTypeToTransformerMapper(mimeTypeToTransformerMapper);
+        frameworkProperties.setValidationQueryFactory(new ValidationQueryFactory(new GeotoolsFilterAdapterImpl(),
+                new GeotoolsFilterBuilder()));
+    }
+
+    @Test
+    public void testGetResource()
+            throws ResourceNotFoundException, IOException, ResourceNotSupportedException,
```

Comment:
Test methods can just throw Exception instead of listing out all the exception they can throw